DeepDiveLogCompiler.scala

```diff
@@ -235,147 +281,168 @@ object DeepDiveLogCompiler extends DeepDiveLogHandler {
   type CompiledBlocks = List[CompiledBlock]
 
   // Dispatch to the corresponding compile function
-  def compile(stmt: Statement, ss: CompilationState): CompiledBlocks = stmt match {
-    case s:  ExtractionRule   => compile(s, ss)
-    case s:  FunctionCallRule => compile(s, ss)
-    case s:  InferenceRule    => compile(s, ss)
-    case _                    => List()  // defaults to compiling into empty block
-  }
+  // def compile(stmts: List[Statement], ss: CompilationState, isIncremental: Boolean): CompiledBlocks = stmts(0) match {
+  //   case s: ExtractionRule   => compileE(stmts, ss, isIncremental)
+  //   case s: FunctionCallRule => compileF(stmts, ss, isIncremental)
+  //   case s: InferenceRule    => compileI(stmts, ss, isIncremental)
+  //   case _                   => List()  // defaults to compiling into empty block
+  // }
 
   // Generate extraction rule part for deepdive
-  def compile(stmt: ExtractionRule, ss: CompilationState): CompiledBlocks = {
-    // Generate the body of the query.
-    val qs              = new QuerySchema( stmt.q )
-    // variable columns
-    val variableCols = stmt.q.head.terms flatMap {
-      case(Variable(v,rr,i)) => ss.resolveColumn(v, qs, stmt.q, true)
-    }
+  def compileE(stmts: List[ExtractionRule], ss: CompilationState, isIncremental: Boolean): CompiledBlocks = {
+    var inputQueries = new ListBuffer[String]()
+    for (stmt <- stmts) {
+      // Generate the body of the query.
+      val qs              = new QuerySchema( stmt.q )
+      // variable columns
+      val variableCols = stmt.q.head.terms flatMap {
+        case(Variable(v,rr,i)) => ss.resolveColumn(v, qs, stmt.q, true)
+      }
+
+      val variableColsStr = if (variableCols.length > 0) Some(variableCols.mkString(", ")) else None
 
-    val variableColsStr = if (variableCols.length > 0) Some(variableCols.mkString(", ")) else None
+      val selectStr = (List(variableColsStr) flatMap (u => u)).mkString(", ")
```

Comment:
I guess this isn't really new code, but it seems like a very convoluted way to generate the `selectStr`.  I think without `variableColsStr`, it's really just:

    val selectStr = if (variableCols.length > 0) variableCols.mkString(", ") else ""

Or even just:

    val selectStr = variableCols.mkString(", ")

Right?